doc/adrs/160_add_support_for_styled_labels_in_view_model_for_trees.adoc

```diff
@@ -0,0 +1,179 @@
+  children: TreeItemLabelElementDescription[1..*]
+}
+
+ReusedTreeItemLabelElementDescription extends TreeItemLabelElementDescription {
+  elements: TreeItemLabelElementDescription[0..*] (non containment reference)
+}
+
+TreeItemLabelFragmentDescription extends TreeItemLabelElementDescription {
+  labelExpression: String
+  style: TextStyleDescription [0..1]
+}
+
+TextStyleDescription {
+  name: Identifier
+  foregroundColorExpression: InterpretedExpression<String>
+  backgroundColorExpression: InterpretedExpression<String>
+  isUnderlineExpression: InterpretedExpression<boolean>
+  isBoldExpression: InterpretedExpression<boolean>
+  isItalicExpression: InterpretedExpression<boolean>
+}
+```
+
+The class `TreeItemLabelElementDescription` is an abstract class that allows to handle the partitioning control flow.
+For instance, optional fragments could be handled by the `IfTreeItemLabelElementDescription` subclass and repeated fragments could be handled by `ForTreeItemLabelElementDescription`.
+The `ReusedTreeItemLabelElementDescription` will handle recursive usage same fragments.
+In this case, references fragment descriptions are not contained and should be defined in other part of the description.
+Ultimately, the fragment description class holds both the label and its text style to compose the styled text part.
+The style property is a non containment reference of a `TextStyleDescription`.
+
+The above style description class `TextStyleDescription`, allows to specify a set of text style details.
+It is named for clarity sake and referenced by the `style` property of the `TreeItemLabelFragmentDescription` class.
+All style descriptions are stored in a new container in the root of the View model called `TextStylePalette`. +
+The ``fontNameExpression`` should return the name of the font to use.
+The color of the label fragment is specified through both ``foregroundColorExpression`` and ``backgroundColorExpression``.
+The string returned by those expressions should be compliant with a CSS color (e.g. ``#rrggbb``).
+The ``isUnderlineExpression`` should return whether this fragment is underlined or not.
+By design, we narrow the expressivity of the underline style to a solid line with the same color as expressed by the ``foregroundColorExpression`` expression.
+The ``isBoldExpression`` and ``isItalicExpression`` expressions respectively control the bold and italic style of this fragment.
+Those three last expressions should be evaluated in a boolean.
+
+=== Usage example
+
+The following style description handles the operation with argument' and returned types in blue.
+
+```
+TreeDescription {
+  treeItemLabelDescriptions: [
+    TreeItemLabelDescription {
+      name: "Operation"
+      preconditionExpression: "aql:self.eClass() = papaya::Operation"
+      children: [
+        TreeItemLabelFragmentDescription {
+          labelExpression: "aql:self.name"
+          style: TextStyleDescription {
```

Comment:
I guess here it would be more a reference to a BackLabelStyle TextStyleDescription

Reply:
Done